infrastructure/network/netadapter/server/grpcserver/grpc_server.go

```diff
@@ -115,23 +116,23 @@ func (s *gRPCServer) handleInboundConnection(ctx context.Context, stream grpcStr
 		return err
 	}
 
-	log.Infof("%s Incoming connection from %s", s.name, peerInfo.Addr)
+	log.Infof("%s Incoming connection from %s #%d", s.name, peerInfo.Addr, connectionCount)
 
 	<-connection.stopChan
-	s.decrementInboundConnectionCount()
 	return nil
 }
 
-func (s *gRPCServer) incrementInboundConnectionCountAndLimitIfRequired() error {
+func (s *gRPCServer) incrementInboundConnectionCountAndLimitIfRequired() (int, error) {
 	s.inboundConnectionCountLock.Lock()
 	defer s.inboundConnectionCountLock.Unlock()
 
 	if s.maxInboundConnections > 0 && s.inboundConnectionCount == s.maxInboundConnections {
-		return errors.Errorf("limit of %d inbound connections has been exceeded", s.maxInboundConnections)
+		log.Warnf("Limit of %d inbound connections has been exceeded", s.maxInboundConnections)
```

Comment:
This is pretty normal, so I wouldn't use `Warnf`. `Debugf` Seems to fit.

Reply:
it is ultimately critical for mining nodes. as you can see this msg only in miner or in debug log mode, which is not an option on mining node...

Reply:
I think it is important for a node operator to notice he has too many RPC clients